homeassistant/components/zwave_js/binary_sensor.py

```diff
@@ -335,58 +285,58 @@ class ZWaveNotificationBinarySensor(ZWaveBaseEntity, BinarySensorEntity):
     """Representation of a Z-Wave binary_sensor from Notification CommandClass."""
 
     def __init__(
-        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+        state_key: str,
     ) -> None:
         """Initialize a ZWaveNotificationBinarySensor entity."""
         super().__init__(config_entry, client, info)
+        self.state_key = state_key
         # check if we have a custom mapping for this value
         self._mapping_info = self._get_sensor_mapping()
 
     @property
     def is_on(self) -> bool:
         """Return if the sensor is on or off."""
-        if self._mapping_info:
-            return self.info.primary_value.value in self._mapping_info["states"]
-        return bool(self.info.primary_value.value != 0)
+        return self.info.primary_value.value is int(self.state_key)
```

Comment:
does `is` work here? I've only seen that used for `None`

Reply:
Yeah, otherwise mypy was yelling at me ;-)

Reply:
Nah, integers, floats, strings etc are compared with `==`. What was the mypy error?

Reply:
Let me check, I've never seen it before

Reply:
Returning Any from function declared to return "bool"  [no-any-return]

Reply:
casted the value to int too and now it's happy